### PR TITLE
Fixed the link to otel semantic conventions docs

### PR DESCRIPTION
**Pull request by natikgadzhi**
The [original link 404s now](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions). They've moved the semantic conversions page out of the specification repo it seems, but there's a [nice doc on the website](https://opentelemetry.io/docs/specs/semconv/general/trace/) — so I've swapped the link to point to that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Then, you can depend on the extras package:
 #### Usage
 
 The OpenTelemetry Semantic Conventions package provides span attributes to work with [OpenTelemetry's
-semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions).
+semantic conventions](https://opentelemetry.io/docs/specs/semconv/general/trace/).
```
Comment (slashmo) [inline README.md:28]:
```suggestion
semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.11.0/specification/trace/semantic_conventions).
```

Both the old and your suggested link link to the latest version of semantic conventions. Currently, `OpenTelemetrySemanticConventions` wraps the attributes defined in v1.11.0 specifically, so how about we link to the tagged conventions as well?

**Review by slashmo — APPROVED** [review]
Thanks for bringing this up, @natikgadzhi! What do you think about using a versioned link instead?

**Comment by natikgadzhi** [comment]
@slashmo, thank you for taking a look, and yep, you're right — versioned is better, gives us more certainty that this one is not going anywhere.

**Comment by ktoso** [comment]
We have new CI now, retriggering

**Pull request merged**